patches/server/0934-Add-paper-dumplisteners-command.patch

```diff
@@ -0,0 +1,90 @@
++++ b/src/main/java/io/papermc/paper/command/PaperCommand.java
+@@ -2,6 +2,7 @@ package io.papermc.paper.command;
+ 
+ import io.papermc.paper.command.subcommands.ChunkDebugCommand;
+ import io.papermc.paper.command.subcommands.DumpItemCommand;
++import io.papermc.paper.command.subcommands.DumpListenersCommand;
+ import io.papermc.paper.command.subcommands.EntityCommand;
+ import io.papermc.paper.command.subcommands.FixLightCommand;
+ import io.papermc.paper.command.subcommands.HeapDumpCommand;
+@@ -50,6 +51,7 @@ public final class PaperCommand extends Command {
+         commands.put(Set.of("syncloadinfo"), new SyncLoadInfoCommand());
+         commands.put(Set.of("dumpitem"), new DumpItemCommand());
+         commands.put(Set.of("mobcaps", "playermobcaps"), new MobcapsCommand());
++        commands.put(Set.of("dumplisteners"), new DumpListenersCommand());
+ 
+         return commands.entrySet().stream()
+             .flatMap(entry -> entry.getKey().stream().map(s -> Map.entry(s, entry.getValue())))
+diff --git a/src/main/java/io/papermc/paper/command/subcommands/DumpListenersCommand.java b/src/main/java/io/papermc/paper/command/subcommands/DumpListenersCommand.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..a93d71102e392ee00dcb238f16b8cdb0ffd412d2
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/command/subcommands/DumpListenersCommand.java
+@@ -0,0 +1,57 @@
++package io.papermc.paper.command.subcommands;
++
++import io.papermc.paper.command.PaperSubcommand;
++import net.kyori.adventure.text.Component;
++import net.kyori.adventure.text.format.NamedTextColor;
++import net.minecraft.server.MinecraftServer;
++import org.bukkit.command.CommandSender;
++import org.bukkit.event.HandlerList;
++import org.checkerframework.framework.qual.DefaultQualifier;
++import org.jetbrains.annotations.NotNull;
++
++import java.util.Comparator;
++import java.util.stream.Stream;
++
++@DefaultQualifier(NotNull.class)
++public class DumpListenersCommand implements PaperSubcommand {
++    @Override
++    public boolean execute(CommandSender sender, String subCommand, String[] args) {
++        this.doDumpListeners(sender, args);
++        return true;
++    }
++
++    public void doDumpListeners(CommandSender sender, String[] args) {
++        if (args.length == 0) {
++            sender.sendMessage(Component.text("Usage: /paper dumplisteners [className]", NamedTextColor.RED));
++            return;
++        }
++
++        try {
++            HandlerList handlers = (HandlerList) Class.forName(args[0]).getMethod("getHandlerList").invoke(null);
++
++            if (handlers.getRegisteredListeners().length == 0) {
++                sender.sendMessage(Component.text(args[0] + " does not have any registered listeners."));
++                return;
++            }
++
++            Component listeners = Stream.of(handlers.getRegisteredListeners())
++                .sorted(Comparator.comparingInt(listener -> listener.getPriority().getSlot()))
```

Comment:
This should not be required. The registered listeners are already sorted whenever they are baked.
Sorting here does not really accomplish anything.

Reply:
I noticed that too after I already created the PR, I assumed it'd be better to keep it in since anyone reading it will know that it's supposed to be sorted high -> low or w/e thanks to the added verbosity.

Reply:
It's better to add a comment then pointless code